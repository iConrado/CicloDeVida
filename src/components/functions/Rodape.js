import React from 'react';
import { View, Text, TouchableOpacity } from 'react-native';
import styles from './styles';

// Rodapé padrão. EXPORT PADRÃO
// PROPS
// children     = componente de texto com style rodape para renderizar as alteraçõe 
//                do comprometimento de renda
// funcProxTela = função do componente originador da chamada para mudar a tela ao clicar no botão 
//                (deve possuir bind)
// tela         = nome da tela a ser chamada

export default props => {
  const { valor, funcProxTela, tela } = props;
  return (
    <View style={styles.viewRodape}>
      <View style={styles.viewRodapeResumo}>
        <View style={styles.viewRodapeResumoLabel}>
          <Text style={styles.rodape}>Comprometimento de renda atual:</Text>
        </View>
        <View style={styles.viewRodapeResumoValor}>
          <Text style={styles.rodape}>{valor}%</Text>
        </View>
      </View>

      <View style={styles.viewRodapeBotao}>
        <TouchableOpacity 
          style={styles.botao}
          onPress={() => funcProxTela(tela)}
        >
          <Text style={styles.txtBotao}>PRÓXIMA ETAPA</Text>
        </TouchableOpacity>
      </View>
    </View>
  );
};

// export default class Rodape extends React.Component {
//   constructor(props) {
//     super(props);
//     this.funcao = _.debounce(this.funcao, 250);
//   }

//   funcao() {
//     const { funcProxTela, tela } = this.props;
//     funcProxTela(tela);
//   }

//   render() {
//     const { valor } = this.props;
//     return (
//       <View style={styles.viewRodape}>
//         <View style={styles.viewRodapeResumo}>
//           <View style={styles.viewRodapeResumoLabel}>
//             <Text style={styles.rodape}>Comprometimento de renda atual:</Text>
//           </View>
//           <View style={styles.viewRodapeResumoValor}>
//             <Text style={styles.rodape}>{valor}%</Text>
//           </View>
//         </View>

//         <View style={styles.viewRodapeBotao}>
//           <TouchableOpacity 
//             style={styles.botao}
//             onPress={() => {
//               this.funcao();
//             }}
//           >
//             <Text style={styles.txtBotao}>PRÓXIMA ETAPA</Text>
//           </TouchableOpacity>
//         </View>
//       </View>
//     );
//   }
// }
