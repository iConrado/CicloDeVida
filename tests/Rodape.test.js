import 'react-native';
import renderer from 'react-test-renderer';

import React from 'react';
import Rodape from '../src/components/functions/Rodape';

it('renders correctly', () => {
  const rendered = renderer.create(<Rodape />).toJSON();
  expect(rendered).toMatchSnapshot();
});
