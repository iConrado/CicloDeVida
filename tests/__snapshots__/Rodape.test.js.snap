// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderTopColor": "#D0D0D0",
      "borderTopWidth": 1,
      "flexDirection": "row",
      "height": 60,
      "marginTop": 0,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 6,
        "flexDirection": "row",
        "paddingHorizontal": 8,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 7,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#666",
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        Comprometimento de renda atual:
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 3,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#666",
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        %
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 4,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#5C86F5",
          "borderRadius": 2,
          "elevation": 4,
          "height": 35,
          "justifyContent": "center",
          "maxWidth": "90%",
          "minWidth": "80%",
          "opacity": 1,
          "paddingVertical": 4,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#FFF",
            "fontSize": 14,
            "fontWeight": "bold",
            "marginHorizontal": 5,
          }
        }
      >
        PRÃ“XIMA ETAPA
      </Text>
    </View>
  </View>
</View>
`;
